ts/parts/Axis.ts

```diff
@@ -7659,13 +7652,9 @@ interface Axis {
     keepProps: Array<string>;
 }
 
-extend(Axis.prototype, {
-
-    // Properties to survive after destroy, needed for Axis.update (#4317,
-    // #5773, #5881).
-    keepProps: ['extKey', 'hcEvents', 'names', 'series', 'userMax', 'userMin']
-
-});
+// Properties to survive after destroy, needed for Axis.update (#4317,
+// #5773, #5881).
+Axis.prototype.keepProps = ['extKey', 'hcEvents', 'names', 'series', 'userMax', 'userMin'];
```

Comment:
Why is this set separately?

Reply:
This is a prototype property shared by all instances. Alternative would be to add it to the class as a static property.

Reply:
Wouldn't that look cleaner?

Reply:
Yes, I agree. When I tried it, unfortunately tests were failing, and since the issue with the Color class, I do not like to make big restructures. I can give it an other try.

Reply:
The difference is, that you can mask the prototype property by setting a instance property with the same name, but we could work around that with a getter function, if necessary.